### ISSUE TITLE
2024/04/23 19:45:22 failed to get token from keyring: failed to unlock correct collection '/org/freedesktop/secrets/aliases/default'

### ISSUE THREAD

**Issue by RestartFU** [rigormorrtiss/discordo#390]
This error happens whenever logging in.

![image](https://github.com/ayn2op/discordo/assets/45609733/936d528a-f885-404f-86fb-d07151845766)

2024/04/23 19:45:22 failed to get token from keyring: failed to unlock correct collection '/org/freedesktop/secrets/aliases/default'

**Comment by ayn2op** [comment]
Do you have `gnome-keyring` installed? You need to install a keyring software to store the authentication token.

**Comment by jpackardts** [comment]
@ayn2op `ps aux | grep keyring` shows that gnome-keyring is running:
```
... /nix/store/kb74hlfibs7qkypaam7skzgzwhi4vg4y-gnome-keyring-46.1/bin/gnome-keyring-daemon --start --foreground --components=ssh,secrets,pkcs11
```

**Comment by mikwee** [comment]
@jpackardts @ayn2op Same:
`mikwee      7222  0.0  0.0 534024  8948 ?        Ssl  11:49   0:00 /usr/bin/gnome-keyring-daemon --start --foreground --components=secrets`

**Comment by PylotLight** [comment]
Why is this closed? 
I have same issue.

**Comment by hustlerone** [comment]
Yeah, it's incredibly frustrating to use this client because of keyring problems.

**Comment by ayn2op** [comment]
> Yeah, it's incredibly frustrating to use this client because of keyring problems.

If you are unable to get keyring software working, you can always pass the `token` flag to the CLI:

```
discordo --token "..."
```

**Comment by hustlerone** [comment]
yeah but that sucks and leaks my token to .history

**Comment by ayn2op** [comment]
Related: https://github.com/zalando/go-keyring/issues/116

**Comment by krisdybvik** [comment]
+1 on this

**Comment by ayn2op** [comment]
Here are the instructions on how to manually set the token for `gnome-keyring`: https://github.com/ayn2op/discordo#linux
